File: js/nova/io/Console.js
var Console = function () {
	var self = this;
	
	
	this.Console = function () {
		self.this();
	};
	
	this.destroy = function () {
	};
	
	this.writeLine = function () {
		self.writeLine("");
	};
	
	this.writeLine = function (text) {
		self.write(text.concat(new String("\n")));
	};
	
	this.writeLine = function (obj) {
		self.writeLine(obj.toString());
	};
	
	this.writeLine = function (num) {
		self.writeLine(Double.toString(num));
	};
	
	this.writeLine = function (num) {
		self.writeLine(Double.toString(num));
	};
	
	this.writeLine = function (num) {
		self.writeLine(Long.toString(num));
	};
	
	this.writeLine = function (num) {
		self.writeLine(Int.toString(num));
	};
	
	this.writeLine = function (num) {
		self.writeLine(Int.toString(num));
	};
	
	this.writeLine = function (num) {
		self.writeLine(Byte.toString(num));
	};
	
	this.writeLine = function (c) {
		self.writeLine(Char.toString(c));
	};
	
	this.write = function (text) {
		var cText;
		cText = text.chars;
		self.fputs(cText.data, self.stdout);
		self.flushOutput();
	};
	
	this.write = function (obj) {
		self.write(obj.toString());
	};
	
	this.write = function (num) {
		self.write(Double.toString(num));
	};
	
	this.write = function (num) {
		self.write(Double.toString(num));
	};
	
	this.write = function (num) {
		self.write(Long.toString(num));
	};
	
	this.write = function (num) {
		self.write(Int.toString(num));
	};
	
	this.write = function (num) {
		self.write(Int.toString(num));
	};
	
	this.write = function (num) {
		self.write(Byte.toString(num));
	};
	
	this.write = function (c) {
		self.write(Char.toString(c));
	};
	
	this.readInt = function () {
		var s;
		s = self.readLine();
		return parseInt(s);
	};
	
	this.readDouble = function () {
		var s;
		s = self.readLine();
		return parseDouble(s);
	};
	
	this.readChar = function () {
		var c;
		c = self.getchar();
		self.flushInput();
		return c;
	};
	
	this.flushInput = function () {
		self.fseek(self.stdin, 0, self.SEEK_END);
	};
	
	this.flushOutput = function () {
		self.fflush(self.stdout);
	};
	
	this.readLine = function () {
		var line;
		var s;
		line = self.ufgets(self.stdin);
		s = new String(line);
		return s;
	};
	
	this.readPassword = function () {
		var pass;
		self.setEcho(false);
		pass = self.readLine();
		self.setEcho(true);
		self.write("\n");
		return pass;
	};
	
	this.setEcho = function (echo) {
		self.nova_setEcho(echo);
	};
	
	this.clearScreen = function () {
		self.nova_clearScreen();
	};
	
	this.waitForEnter = function () {
		var c;
		self.flushInput();
		c = [];
		self.fgets(c, 2, self.stdin);
	};
	
	this.this = function () {
	};
	
	this.super = function () {
	};
	
	
};

File: java/nova/io/Console.java
package nova.io;

import java.util.Optional;
import nova.exception.ExceptionData;
import nova.exception.Exception;
import nova.exception.DivideByZeroException;
import nova.io.Console;
import nova.primitive.number.Number;
import nova.primitive.number.Byte;
import nova.primitive.number.Short;
import nova.primitive.number.Int;
import nova.primitive.number.Long;
import nova.primitive.number.Float;
import nova.primitive.number.Double;
import nova.primitive.Null;
import nova.primitive.number.Char;
import nova.primitive.Bool;
import nova.datastruct.list.Array;
import nova.datastruct.list.IntArray;
import nova.datastruct.list.CharArray;
import nova.datastruct.list.DoubleArray;
import nova.datastruct.list.IntRange;
import nova.thread.Thread;
import nova.thread.async.Async;
import nova.gc.GC;
import nova.math.Math;
import nova.Object;
import nova.String;
import nova.System;
import nova.Class;

public class NovaConsole
{
	
	
	
	
	
	public NovaConsole()
	{
		init();
	}
	
	
	
	
	
	
	
	
	
	public static void writeLine()
	{
		writeLine(new NovaString(""));
	}
	
	public static void writeLine(NovaString text)
	{
		write(text.concat(new NovaString("\n")));
	}
	
	public static void writeLine(NovaObject obj)
	{
		writeLine(obj.toString());
	}
	
	public static void writeLine(double num)
	{
		writeLine(NovaDouble.toString(num));
	}
	
	public static void writeLine(float num)
	{
		writeLine(NovaDouble.toString(num));
	}
	
	public static void writeLine(long num)
	{
		writeLine(NovaLong.toString(num));
	}
	
	public static void writeLine(int num)
	{
		writeLine(NovaInt.toString(num));
	}
	
	public static void writeLine(short num)
	{
		writeLine(NovaInt.toString(num));
	}
	
	public static void writeLine(byte num)
	{
		writeLine(NovaByte.toString(num));
	}
	
	public static void writeLine(char c)
	{
		writeLine(Char.toString(c));
	}
	
	public static void write(NovaString text)
	{
		CharArray cText;
		cText = text.chars;
		flushOutput();
	}
	
	public static void write(NovaObject obj)
	{
		write(obj.toString());
	}
	
	public static void write(double num)
	{
		write(NovaDouble.toString(num));
	}
	
	public static void write(float num)
	{
		write(NovaDouble.toString(num));
	}
	
	public static void write(long num)
	{
		write(NovaLong.toString(num));
	}
	
	public static void write(int num)
	{
		write(NovaInt.toString(num));
	}
	
	public static void write(short num)
	{
		write(NovaInt.toString(num));
	}
	
	public static void write(byte num)
	{
		write(NovaByte.toString(num));
	}
	
	public static void write(char c)
	{
		write(Char.toString(c));
	}
	
	public static int readInt()
	{
		NovaString s;
		s = readLine();
		return NovaInt.parseInt(s);
	}
	
	public static double readDouble()
	{
		NovaString s;
		s = readLine();
		return NovaDouble.parseDouble(s);
	}
	
	public static char readChar()
	{
		char c;
		c = getchar();
		flushInput();
		return c;
	}
	
	private static void flushInput()
	{
	}
	
	private static void flushOutput()
	{
	}
	
	public static NovaString readLine()
	{
		char[] line;
		NovaString s;
		s = new NovaString(line);
		return s;
	}
	
	public static NovaString readPassword()
	{
		NovaString pass;
		setEcho(false);
		pass = readLine();
		setEcho(true);
		write(new NovaString("\n"));
		return pass;
	}
	
	public static void setEcho(boolean echo)
	{
		nova_setEcho(echo);
	}
	
	public static void clearScreen()
	{
		nova_clearScreen();
	}
	
	public static void waitForEnter()
	{
		char[] c;
		flushInput();
		c = new char[2];
	}
	
	public void init()
	{
	}
	
}
